CONTRIBUTING.md

```diff
@@ -1,26 +1,43 @@
-# Contributing Guidelines
+# Contribution guidelines
 
-Welcome to Kubernetes. We are excited about the prospect of you joining our [community](https://git.k8s.io/community)! The Kubernetes community abides by the CNCF [code of conduct](code-of-conduct.md). Here is an excerpt:
+## Sign the CLA
 
-_As contributors and maintainers of this project, and in the interest of fostering an open and welcoming community, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities._
+Kubernetes projects require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests.  Please see https://git.k8s.io/community/CLA.md for more info
 
-## Getting Started
+## Contributing a Patch
 
-We have full documentation on how to get started contributing here:
+1. Submit an issue describing your proposed change to the repo in question.
+1. The [repo owners](OWNERS) will respond to your issue promptly.
+1. If your proposed change is accepted, and you haven't already done so, sign a Contributor License Agreement (see details above).
+1. Fork the desired repo, develop and test your code changes.
+1. Submit a pull request.
 
-<!---
-If your repo has certain guidelines for contribution, put them here ahead of the general k8s resources
--->
+## Development
+Please go through [CSI Spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) and [General CSI driver development guideline](https://kubernetes-csi.github.io/docs/developing.html) to develop a basic understanding of CSI driver before you start.
 
-- [Contributor License Agreement](https://git.k8s.io/community/CLA.md) - Kubernetes projects require that you sign a Contributor License Agreement (CLA) before we can accept your pull requests
-- [Kubernetes Contributor Guide](https://k8s.dev/guide) - Main contributor documentation, or you can just jump directly to the [contributing page](https://k8s.dev/docs/guide/contributing/)
-- [Contributor Cheat Sheet](https://k8s.dev/cheatsheet) - Common resources for existing developers
+### Requirements
+* Golang 1.15.+
+* [Ginkgo](https://github.com/onsi/ginkgo) in your PATH for end-to-end testing
+* Docker 17.05+ for releasing
 
-## Mentorship
+### Dependency
+Dependencies are managed through go module. To build the project simply type: `make`
 
-- [Mentoring Initiatives](https://k8s.dev/community/mentoring) - We have a diverse set of mentorship programs available that are always looking for volunteers!
+### Testing
+* To execute all unit tests, run: `make test`
+* To execute sanity test run: `make test-sanity`
 
-## Contact Information
+### Release Process
+Please see [Release Process](./docs/release.md).
```

Comment:
Also doesn't exist yet, But I don't see a ton of harm to leaving this in right now so we don't forget to re-add it when the release process is finalized.